src/main/java/pl/onsight/wypozyczalnia/service/ProductOrderService.java

```diff
@@ -17,4 +17,9 @@
     List<ProductOrderEntity> findUserOrders(Long id);
 
     void removeProductOrder(Long id);
+
+    ProductOrderEntity getOrderById(Long id);
+
+    boolean isUserHavePermissionToSeeThisOrder(Long userId, Long orderId);
+
```

Comment:
wydaje mi się że metoda bardziej pasuje do validatora

Reply:
Coś w tym jest. Powstał UserValidator, bo do orderValidatora mi to nie pasowało :P